onnxruntime/test/framework/nuphar/nuphar_single_op_test.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/status.h"
+#include "core/common/logging/logging.h"
+#include "core/framework/execution_provider.h"
+#include "core/framework/op_kernel.h"
+#include "core/framework/session_state.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/graph/graph.h"
+#include "core/graph/model.h"
+#include "core/graph/op.h"
+#include "core/providers/cpu/math/element_wise_ops.h"
+#include "core/session/inference_session.h"
+#include "gtest/gtest.h"
+#include "test/capturing_sink.h"
+#include "test/framework/test_utils.h"
+#include "test/test_environment.h"
+#include "test/util/include/default_providers.h"
+
+using namespace std;
+using namespace ONNX_NAMESPACE;
+
+namespace onnxruntime {
+namespace test {
+
+TEST(NupharTest, PadTest) {
+#ifdef USE_NUPHAR
```

Comment:
Would be helpful before each unit test here and in other files to have a comment stating the purpose of the test when it's not immediately obvious. What is it testing? Why are we verifying that? Is the choice of Pad significant or was that a random op?

Also wondering why this isn't covered by the unit testing for the Pad op, given OpTester knows about the nuphar provider so in theory the Pad tests would run using that anyway.

Reply:
OpTester only test single node, while here the compiler tests the combination. I believe the test here is already covered elsewhere and it can be removed.

---
In reply to: [312425819](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312425819) [](ancestors = 312425819)

Reply:
FWIW you can add multiple nodes by inheriting from OpTester and implementing the AddNodes method. See the control flow tests for examples.

---
In reply to: [312633354](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312633354) [](ancestors = 312633354,312425819)

Reply:
Good information, thanks

---
In reply to: [313220098](https://github.com/microsoft/onnxruntime/pull/881#discussion_r313220098) [](ancestors = 313220098,312633354,312425819)